### DISCUSSION TITLE
Fabric and Webgl

### DISCUSSION THREAD

**Discussion by BirkhoferAntoine** [fabricjs/fabric.js#7347] [Q&A]
Hi, first thank you very much for all the hardwork that made my job's project a lot more bearable and enjoyable!

I have a fabric canvas that needs to handle a lot of images with big sizes as well.
After some time i end up with the error "NS_ERROR_FAILURE" which i think is due to the limitation of memory usage on canvas API.

If i am not mistaken using the Webgl API would allow me to pass this limitation, but i don't find any real straight answer on how to use webgl with fabric and fabricCanvas.getContext('webgl') (or variants) returns null while lambdaCanvas returns true.

Could anybody help me or just give me proper orientation on where to start looking ?
Otherwise if there is a shorter/better way to pass the memory limit i am all ears!
Thanks.

**Comment by asturur** [comment]
Hi @BirkhoferAntoine, is unclear where are you getting this error and doing what.
There are no real limitation of memory usage on canvas, at least none that i ever hit.
With WEBGL everything becomes a bit more fragile and handled at low level, You could eventually load large textures ( on some specific hardware ) but to pull them out you are still limited from the canvas size.

On what OS/BROWSER are you getting this error? for which canvas size?

**Comment by BirkhoferAntoine** [comment]
Hi @asturur, the user is putting a lot of different images of very good quality and the size of the canvas change depending on what the user wants to put on it / the amount etc, but main idea is size big enough to reach memory limit.
I really need to be able to go through it, which is dependent on browsers but more or less the same (exemple even if outdated) :
https://stackoverflow.com/questions/6081483/maximum-size-of-a-canvas-element

But from what you just said even with WebGL i would still be limited to some size even if webgl and the gpu takes over the task ?

**Reply by asturur** [reply]
The webgl is usually smaller than the normal canvas.

**Reply by asturur** [reply]
Also please try to make the error happen on jsfiddle, so we can understand what kind of error we are talking about

**Comment by BirkhoferAntoine** [comment]
Here are the limitations per browser with package to reproduce the test:
https://github.com/jhildenbiddle/canvas-size#test-results
